### ISSUE TITLE
DynamoDBPersistenceLayer should use ExpressionAttributeNames in _put_record

### ISSUE THREAD

**Issue by Tankanow** [awslabs/aws-lambda-powertools-python#693]
**What were you trying to accomplish?**

## Expected Behavior
I should be able to use DynamoDB reserved words as Item attributes. For example, `ttl` is a common DynamoDB Item Time to Live attribute name. A consumer should be able to specify reserved word attribute names for any of the `*_attr` values when configuring `DynamoDBPersistenceLayer`.

## Current Behavior
`_put_record` currently throws an exception b/c it uses Python string interpolation instead of `ExpressionAttributeNames`.

## Possible Solution
Switch to `ExpressionAttributeNames`, e.g. 
```
            self.table.put_item(
                Item=item,
                ConditionExpression='attribute_not_exists(#key) OR #now < :now',
                ExpressionAttributeNames={"#key": self.key_attr, "#now": self.expiry_attr},
                ExpressionAttributeValues={":now": int(now.timestamp())},
            )
```

## Steps to Reproduce (for bugs)
1. Create a DynamoDB table with a Time to Live attribute called `ttl`.
2. Configure a `DynamoDBPersistenceLayer` with `expiry_attr='ttl'`
3. Run

## Environment

* **Powertools version used**: 1.20.2
* **Packaging format (Layers, PyPi)**: PyPi
* **AWS Lambda function runtime:** Python 3.8
* **Debugging logs**

> [How to enable debug mode](https://awslabs.github.io/aws-lambda-powertools-python/#debug-mode)**

```python
# paste logs here
```

**Comment by Tankanow** [comment]
I'm happy to submit a PR for this. I have it locally.

**Comment by heitorlessa** [comment]
Hey @Tankanow, same as the other one -- another oversight from our part, I'd love a PR as it's a quick one.

**Comment by heitorlessa** [comment]
This is now available as part of today's release 1.21.0 @Tankanow ;) Thank you so much for contributing that!

https://github.com/awslabs/aws-lambda-powertools-python/releases/tag/v1.21.0
